### ISSUE TITLE
Show backlinks on publicly shared documents

### ISSUE THREAD

**Issue by kingfungdesign** [outline/outline#4820]
Backlinks are not shown in the publicly shared version of the documents.

**To Reproduce**
1. Enable "Publish to internet" and "Share nested documents" for a document that has internal links to another document.
2. Navigate to the public URL for the shared document.
3. Browse to the document linked inside the page.

**Expected behavior**
There should be a backlink section under the page that shows the document in step 1.

**Screenshots**
![image](https://user-images.githubusercontent.com/52369186/216369338-2d4fd4a5-2ee5-4ffa-ba97-b6e120b84b8f.png)
![image](https://user-images.githubusercontent.com/52369186/216369413-5e0fef11-874d-4ef5-8137-87937bf88b64.png)

**Outline (please complete the following information):**
- Install: getoutline.com
- Version: v0.67.2

**Desktop (please complete the following information):**
 - OS: Windows 10 Pro
 - Browser: Edge Beta 
 - Version: Version 110.0.1587.22 (Official build) beta (64-bit)

If this is not the expected behavior, I'd like to propose to have this as an enhancement. I would want the viewers that I share my knowledge to have the closest browsing experience as I do. Not having the backlinks for viewers makes the knowledge base more disconnected than it has to be.

**Comment by almereyda** [comment]
Can we reopen this? @tommoor 

It appears this is less a cosmetic issue, but rather one about consistency in experience (and expectations).

Maybe it will be possible to toggle this?

Or rather switch to the new default: both views being similar in spatial layout and navigational fuction.

**Comment by vitekr** [comment]
I second the idea of reopening this issue. We use Outline in various instances, serving both as an internal resource and a public knowledgebase. The absence of backlinks in public sections significantly impacts the user experience.

**Comment by almereyda** [comment]
This little inconsistency still persists: Backlinks on publicly shared pages will render navigating them a lot more easy. Obviously this will only work for shared document trees with subdocuments.

**Comment by almereyda** [comment]
This was a nice idea. The data is there. Can we expose it securely, e.g. only for pages which are sub pages to the shared page?

**Comment by tommoor** [comment]
Yep, I don't think there is anything wrong with the idea it's just pretty far down the backlog compared to other features folks are requesting

**Comment by almereyda** [comment]
Which is completely fine.

**Comment by almereyda** [comment]
No, please don't close.

We're just about to redo some publicly shared pages, e.g. with #7660.

Maybe this component will find attention in the next cycle, too?

**Comment by almereyda** [comment]
I didn't see this being closed. This is still not the case.

**Comment by almereyda** [comment]
💐
